nats.go

```diff
@@ -1660,7 +1669,7 @@ func (nc *Conn) connectProto() (string, error) {
 	}
 
 	cinfo := connectInfo{o.Verbose, o.Pedantic, ujwt, nkey, sig, user, pass, token,
-		o.Secure, o.Name, LangString, Version, clientProtoInfo, !o.NoEcho}
+		o.Secure, o.Name, LangString, Version, clientProtoInfo, !o.NoEcho, true}
```

Comment:
I wonder if true should become the header version 1.
And a corresponding check on the server client has header enabled if server supported header version >= client header version.

A client talking header version 2 should probably be downgraded to one without header when the server only speaks version 1.

Reply:
So change it to an into or string? I think I would prefer to add later when we cross that bridge, and if its needed.

Reply:
I had an int in mind, but yeah. 
The problem is if you change to a different version in the future, the then "old" server code, already needs to handle unknown versions.

Reply:
It will since it will be looking for the first line to be NATS/1.0

Reply:
A message from a client sending a v2 header will pass through an old server and can be received by a v1 client who then trips.

compared to.

When you communicate your version to the server it can enable/disable header based on its understanding of header. Essentially what we do today when adding header. If you don't understand them we strip them.

Reply:
I understand, probably will not need this for many years and would be a bunch of work. Also clients should check the version inlined in the header so they don't trip.

Reply:
Meaning we need client support regardless, not sure servers need to be involved.

Reply:
Server need to understand versions as soon as they start reading them. (tracing)
They can provide a migration path by stripping the header if it is beyond what the client/other server can understand.

Reply:
The server will be acting like a client here as well, same logic. Read first line of header and check for `NATS/V.v`